### ISSUE TITLE
unable to get the same confusion matrix shown in Chapter 2 out [22]

### ISSUE THREAD

**Issue by hopezh** [ageron/handson-ml2#42]
In the notebook of Chapter 2 on **confusion matrix**, the out [22] is:
![Screenshot 2019-10-16 at 12 39 23 PM](https://user-images.githubusercontent.com/23435810/66888773-75539400-f012-11e9-80ed-ec13edac4403.png)

However, the output generated in my notebook is:
![Screenshot 2019-10-16 at 12 39 35 PM](https://user-images.githubusercontent.com/23435810/66888798-89979100-f012-11e9-97a2-26b648c318ca.png)

May I ask if I have missed any important step here? 

My notebook is shown here (please use nbviewer if it's not rendered in Github):
[https://github.com/hopezh/learn_hands_on_ML_v2/blob/master/Ch3%20classification.ipynb](url)

Thanks.

**Comment by hopezh** [comment]
Thank you very much, @ageron ,
Yes, you're right that I wrongly set the training set to contain only 6K instances: 

![Image 1](https://user-images.githubusercontent.com/23435810/66975066-cd9c9b80-f0cf-11e9-8a57-0f8683060df7.png)

Once I changed it to 60K, i got the following results from confusion matrix. Although they are similar to the results in your book in terms of magnitude of the values, they are not the same, as shown below. May I ask **if it is expected that we may get slightly different results in this step**? and Why? 

![Image 2](https://user-images.githubusercontent.com/23435810/66975068-d1c8b900-f0cf-11e9-840e-b18cd3f7f7be.png)

**Comment by ageron** [comment]
Hi @hopezh ,
Yes, it's certainly possible to get slightly different results, since the model is stochastic (i.e., training relies on randomness). However, if you run exactly the notebook in the project (without any modification) and you run the cells perfectly in order, then you should normally get exactly the same results as in the book since the notebook sets the random seed. That said, perfect reproducibility is tricky, since everything needs to be perfectly identical. Check out my [video on this topic ](https://www.youtube.com/watch?v=Ys8ofBeR2kA)(it's about reproducibility specifically with TensorFlow, but it makes a number of points that are valid as well for Scikit-Learn or any other library).
Hope this helps!

**Comment by hopezh** [comment]
Thank you very much, @ageron .
